src/main/java/com/google/sps/data/EventUpdater.java

```diff
@@ -36,7 +49,159 @@ public Entity getEntity() {
+      }
+
+      setEventProperty(propertyKey, formValue);
+    }
+
+    setNonFormProperties(forRegistration, historyUpdate);
+
+    setEventDateAndHours(request);
+
+    setEventOwnerOrgNameBasedOnId(ownerOrgId);
+  }
+
+  private void setEventOwnerOrgNameBasedOnId(long ownerOrgId) {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    Filter queryFilter = new FilterPredicate(Entity.KEY_RESERVED_PROPERTY, FilterOperator.EQUAL, ownerOrgId);
+    Query query = new Query("Distributor").setFilter(queryFilter);
+    PreparedQuery preparedQuery = datastore.prepare(query);
+
+    Entity entity = null;
+    try {
+      entity = preparedQuery.asSingleEntity();
+    } catch(PreparedQuery.TooManyResultsException exception) {
+      logger.log(Level.SEVERE, "Multiple Distributor entities found with ID: " + Long.toString(ownerOrgId) + ".");
+    }
+
+    String ownerOrgName = (String) entity.getProperty("orgName");
+    this.entity.setProperty("eventOwnerOrgName", ownerOrgName);
+  }
+
+  private String getParameterOrThrow(HttpServletRequest request, String formKey) {
+    String result = request.getParameter(formKey);
+    if (result == null || result.isEmpty()) {
+      throw new IllegalArgumentException();
+    }
+    return result;
+  }
+
+  private ArrayList<String> getParameterValuesOrThrow(HttpServletRequest request, String formKey){
+    ArrayList<String> results = new ArrayList<String>(Arrays.asList(request.getParameterValues(formKey)));
+    if (results.isEmpty() || results == null) {
+      throw new IllegalArgumentException("Form value cannot be null");
+    }
+
+    // Checks if there is a value that is empty which means a blank time range was submitted
+    for(int i = 0; i < results.size(); i++) {
+      if(results.get(i).equals("")) {
+        throw new IllegalArgumentException("Form value cannot be null");
+      }
+    }
+    return results;
+  }
+
+  // Sets form values based on property key.
+  private void setEventProperty(String propertyKey, String formValue) {
+    if (propertyKey.equals("eventPartnerNames")) {
+      // Stores partnering organizations's names; partnering organizations do not have the ability to edit Event, so there is no need to store org IDs.
+      ArrayList<String> parsedNames = new ArrayList<String>(Arrays.asList(formValue.split("\\s*,\\s*")));
+
+      this.entity.setProperty(propertyKey, parsedNames);
+      return;
+    }
+
+    this.entity.setProperty(propertyKey, formValue);
   }
+
+  // Sets values not passed in through the request form such as creation time of Event and last edited time.
+  private void setNonFormProperties(boolean forRegistration, EmbeddedEntity historyUpdate) {
+    long milliSecondsSinceEpoch = (long) historyUpdate.getProperty("changeTimeStampMillis");
+    ArrayList<EmbeddedEntity> changeHistory = new ArrayList<EmbeddedEntity>();
+
+    if (forRegistration) {
+      this.entity.setProperty("eventCreationTimeStampMillis", milliSecondsSinceEpoch);
```

Comment:
can you remove the 'event' from the beginning of the timestamp properties?

Reply:
yes!

Reply:
Doesn't look like the change happened?